src/Chapter3.hs

```diff
@@ -910,6 +967,18 @@ Implement instances of "Append" for the following types:
 class Append a where
     append :: a -> a -> a
 
+newtype Gold = Gold Int
+
+instance Append Gold where
+  append (Gold g1) (Gold g2) = Gold $ g1 + g2
+
+instance Append [a] where
+  append = (++)
+
+instance (Append a) => Append (Maybe a) where
+  append m1 m2  = append <$> m1 <*> m2
```

Comment:
While this is valid implementation and nothing in the task description says differently, in reality it would be more beneficial is this function returned `Just` when at least one argument is `Just`.